tools/src/main/java/com/graphhopper/tools/Measurement.java

```diff
@@ -128,25 +128,37 @@ protected void prepareCH(boolean closeEarly) {
                 // note that we measure the total time of all (possibly edge&node) CH preparations
                 put(Parameters.CH.PREPARE + "time", sw.stop().getMillis());
                 int edges = getGraphHopperStorage().getEdges();
-                if (!getCHPreparationHandler().getNodeBasedCHProfiles().isEmpty()) {
-                    CHProfile chProfile = getCHPreparationHandler().getNodeBasedCHProfiles().get(0);
+                int i = 0;
```

Comment:
I think I would prefer to keep it simple here and just deal with one profile per measurement. So running the measurement for a custom profile would just mean running the same class again with different parameters. For example we could add a program argument that we can use to pass the path to a custom model file. It would also be more flexible this way and we'd get all the measurements automatically  (`routing`,`routing_CH`,`routing_CH_no_sod`, ....) for this profile. For the benchmarks this should also be no problem, because we can just assign a different measurement name to the resulting files so we can easily separate the results from those we obtain for he 'standard' profile. If you agree I can set this up.

Reply:
This is a good idea. In fact I already faced a disadvantage of the used approach: more CH profiles means more memory is used and with an unchanged -Xmx setting even the speed of the unchanged "fastest CH" degraded (probably GC had to be more active slowing down everything else a bit)

Reply:
Ah yes good point. Maybe its also slower because of extra junctions when multiple vehicles are involved? I like calling them 'ghost junctions' (in my head) btw :) I mean those that cause #1858.

Reply:
Ok will do this is also a nice playground to get some practice using the custom profiles :)

Reply:
> Maybe its also slower because of extra junctions when multiple vehicles are involved?

Unlikely, as both are derived from car. But I don't know for sure :)